### DISCUSSION TITLE
Compatibility of msgpack between ArduinoJson6 and nlohmann json

### DISCUSSION THREAD

**Discussion by HappySamuel** [nlohmann/json#2858] [Q&A] [answered]
Hi

I would like to ask if is there any compatibility issue in msgpack exchanging between ArduinoJson6 and nlohmann json? For example: 
Embedded Code (using ArduinoJson6)  <----- msgpack -----> PC (using nlohmann json)

Is this example possible? Or has anyone did this before?

Best,
Samuel

**Comment by nlohmann** [comment, marked as answer]
This should work. See https://json.nlohmann.me/features/binary_formats/messagepack/ for all details on the MsgPack implementation.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Comment by HappySamuel** [comment]
Sorry for the late reply. Yes i have tried ArduinoJson v6 and nlohmann json on the msgpack communication. They are compatible with each other. 

Best,
Samuel

**Comment by HappySamuel** [comment]
Hi @nlohmann 

I have following questions:
(1) Is there a way to know how much total byte size of msgpack per stream that generated through nlohmann::json::to_msgpack() ?
(2) Is there a way to know how much total byte size msgpack per stream that received through nlohmann::json::from_msgpack() ?

Best,
Samuel

**Reply by nlohmann** [reply]
No, the library does not track this.
